.github/workflows/main.yml

```diff
@@ -0,0 +1,118 @@
+name: Run Tests and/or Deploy to S3
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+    # NOTE: ANY branches listed here will be deployed to due to `should_deploy` logic.
+    # - master # TODO: Enable when merging into master.
+    - develop
+  # schedule:
+    # This will be ran against [default branch = master]
+    # TODO: Enable when merging into master.
+    # - cron: 0 */8 * * * # every 8 hours
```

Comment:
should we make this more frequent? Ideally we'll have a check for updates, but even before then maybe once every 2 hours? 

Are there any limits on how often we can trigger the github actions build?

Reply:
This could be a lot more frequent.  Since it's open source, I don't think we pay for these builds...Github's pricing is actually a bit unclear still too, given it's beta.

Since we're still pushing commits in from Heroku, I think 8 hours is even overkill, but for the Helpsite, which has no Heroku, every hour is probably fine.

---

As far as checking for updates with Contentful, there's a `sync` API, but that requires you to maintain a `sync_token`.  I can't see a good way to actually get "all changes since last build" in general, querying with a timestamp, but even if we could, I don't know where we'd store that timestamp.  I fear builds would be dropped from time to time.

https://help.github.com/en/articles/events-that-trigger-workflows#external-events-repository_dispatch is still the best bet..  That page's documentation seems to state you can target which branch gets built (which would be what we need), but I can't see how via the API documentation: https://developer.github.com/v3/repos/#create-a-repository-dispatch-event

Reply:
Can we just send the contentful webhook request direct to github and trigger it that way? That would allow us to cut out buildgoat at least, although it may also spam too many builds...

Reply:
Oh, or is it just *github* webhook events that it can use :(